### PR TITLE
Installation script to devtron main repo - automated release parameters - grafana requests through devtron

### PR DESCRIPTION
**Pull request by prakarsh-dt**
# Description

- Brought [devtron-installation-script](https://github.com/devtron-labs/devtron-installation-script) into devtron main repo in manifests folder.
- Added parameters in installation script for easier automated releases of devtron micro-services. 
- Grafana requests routed through devtron Fixes # (https://github.com/devtron-labs/devtron/issues/343)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Has been tested by creating a new cluster and installing Devtron using helm3 and kubectl on AWS and Azure clusters, here is the complete list of test cases

cloud | blob storage | installed via | app deploy (view grafana dashboard) | chart deploy
-- | -- | -- | -- | --
AWS | s3 | helm | Done | Done
AWS | s3 | kubectl | Done | Done
AWS | Minio | helm | Done | Done
AWS | Minio | kubectl | Done | Done
azure | azure blob | helm | Done | Done
azure | azure blob | kubectl | Done | Done
azure | Minio | helm | Done | Done
azure | Minio | kubectl | Done | Done

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have tested it for all user roles

### CODE REVIEW

**Comment by nishant-d** [comment]
we need to test it for more use cases before releasing it in public. 
some of the use cases that i can think of are :  
https://docs.google.com/spreadsheets/d/14dcI_pVjfVuXWlbhh90wrBEG55K3SEMfa9HQz0GV-Vc/edit?usp=sharing

**Comment by pghildiyal** [comment]
Test cases for grafana
1. Add prometheus url to cluster and check if it got registered in grafana
2. Check, application metrics should be available against application

File: manifests/yamls/minio-storage.yaml [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+kind: Secret
+metadata:
+  name: devtron-minio
+  labels:
+    app: minio
+    chart: minio-8.0.10
+    release: devtron
+    heritage: Helm
+type: Opaque
+data:
```
Comment (pghildiyal) [inline manifests/yamls/minio-storage.yaml:35]: remove this

File: manifests/yamls/devtron.yaml [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+type: Opaque
+---
+# Source: devtron/templates/configmap.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: devtron-cm
+  labels:
+    release: devtron
+data:
```
Comment (pghildiyal) [inline manifests/yamls/devtron.yaml:17]: add GRAFANA_HOST, GRAFANA_PORT, GRAFANA_NAMESPACE

**Review by nishant-d — APPROVED** [review]

**Pull request merged**